### PR TITLE
feat: add QR code renderer and index renderer

### PR DESCRIPTION
**Pull request by MichaelTaylor3D**

### CODE REVIEW

File: src/server/Controllers/StoresController.cs [hunk 1]
```diff
@@ -139,6 +139,14 @@ public async Task<IActionResult> GetStoreCatchAll(string storeId, string catchAl
 
             if (!string.IsNullOrEmpty(fileExtension))
             {
+                if (fileExtension == ".offer")
```
Comment (MichaelTaylor3D) [inline src/server/Controllers/StoresController.cs:142]: ideally we create a mimtype map, that dynamically chooses the renderer to use. I might need your help on how to do that in dotnet though.

File: src/server/Renderers/OfferFile.cs [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+{
+    public string RenderOffer(string offerFile)
+    {
+        var qrGenerator = new QRCodeGenerator();
+        var qrCodeData = qrGenerator.CreateQrCode(offerFile, QRCodeGenerator.ECCLevel.L);
+        var qrCode = new PngByteQRCode(qrCodeData);
+        var qrCodeImage = qrCode.GetGraphic(20);
+
+        var base64Image = Convert.ToBase64String(qrCodeImage);
+        string qrCodeImg = $"<a href=\"data:text/plain;charset=utf-8,{Uri.EscapeDataString(offerFile)}\" download=\"offerFile.txt\"><img src=\"data:image/png;base64,{base64Image}\" class=\"qr-image\" alt=\"QR Code\" title=\"Tap to download offer\" /></a>";
```
[the remark this answers is not in the document]
Comment (MichaelTaylor3D) [reply src/server/Renderers/OfferFile.cs]: factored this out for the RenderFactory

File: src/server/Controllers/StoresController.cs (hunk 1 above)
Comment (dkackman) [reply src/server/Controllers/StoresController.cs]: I've already integrated file extension to mime type library, Could just build a dictionary of other mimetypes to look at if that library passes on em.

look here:
https://github.com/Datalayer-Storage/Distributed-Internet-Gateway/blob/fb4702f941862ad6f0d6a9e18e3c4150ab79f75f/src/server/Controllers/StoresController.cs#L145-L146

You could add a static dictionary like so for those other file types
```
    private static readonly Dictionary<string, string> otherMimeTypes = new()
    {
        { "offer", "application/json"}
    };
```

File: src/server/Renderers/OfferFile.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (dkackman) [reply src/server/Renderers/OfferFile.cs]: coolio - I'll merge the PR and if there's anything that can be done in a more asp dotnet way ill follow on.

**Pull request merged**